src/org/opendatakit/briefcase/ui/export/ExportConfiguration.java

```diff
@@ -0,0 +1,177 @@
+import static org.opendatakit.briefcase.ui.MessageStrings.DIR_INSIDE_ODK_DEVICE_DIRECTORY;
+import static org.opendatakit.briefcase.ui.MessageStrings.DIR_NOT_DIRECTORY;
+import static org.opendatakit.briefcase.ui.MessageStrings.DIR_NOT_EXIST;
+import static org.opendatakit.briefcase.ui.StorageLocation.isUnderBriefcaseFolder;
+import static org.opendatakit.briefcase.util.FileSystemUtils.isUnderODKFolder;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public class ExportConfiguration {
+  private Optional<Path> exportDir;
+  private Optional<Path> pemFile;
+  private Optional<LocalDate> startDate;
+  private Optional<LocalDate> endDate;
+
+  private ExportConfiguration(Optional<Path> exportDir, Optional<Path> pemFile, Optional<LocalDate> startDate, Optional<LocalDate> endDate) {
+    this.exportDir = exportDir;
+    this.pemFile = pemFile;
+    this.startDate = startDate;
+    this.endDate = endDate;
+  }
+
+  public static ExportConfiguration empty() {
+    return new ExportConfiguration(Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
+  }
+
+  public static ExportConfiguration copy(ExportConfiguration other) {
+    return new ExportConfiguration(
+        other.exportDir,
+        other.pemFile,
+        other.startDate,
+        other.endDate
+    );
+  }
+
+  public Optional<Path> getExportDir() {
+    return exportDir;
+  }
+
+  public Optional<Path> getPemFile() {
+    return pemFile;
+  }
+
+  public Optional<LocalDate> getStartDate() {
+    return startDate;
+  }
+
+  public void setStartDate(LocalDate date) {
+    this.startDate = Optional.ofNullable(date);
+  }
+
+  public Optional<LocalDate> getEndDate() {
+    return endDate;
+  }
+
+  public void setEndDate(LocalDate date) {
+    this.endDate = Optional.ofNullable(date);
+  }
+
+  private boolean isDateRangeValid() {
+    return !startDate.isPresent() || !endDate.isPresent() || startDate.get().isBefore(endDate.get());
+  }
+
+  public void setExportDir(Path path) {
+    this.exportDir = Optional.ofNullable(path);
+  }
+
+  public void setPemFile(Path path) {
+    this.pemFile = Optional.ofNullable(path);
+  }
+
+  public void ifExportDirPresent(Consumer<Path> consumer) {
+    exportDir.ifPresent(consumer);
+  }
+
+  public void ifPemFilePresent(Consumer<Path> consumer) {
+    pemFile.ifPresent(consumer);
+  }
+
+  public void ifDateRangeStartPresent(Consumer<LocalDate> consumer) {
+    startDate.ifPresent(consumer);
+  }
+
+  public void ifDateRangeEndPresent(Consumer<LocalDate> consumer) {
+    endDate.ifPresent(consumer);
+  }
+
+  private List<String> getErrors() {
+    List<String> errors = new ArrayList<>();
+
+    if (!exportDir.isPresent())
+      errors.add("Export directory was not specified.");
```

Comment:
Must an export directory be supplied on the custom config dialog? What if the user just wants to override the date range?

Reply:
That's a good point. We've tried to start simple and work on the smallest possible feature set and on this PR we're focusing on the simplest form of "custom configurations per form". 

I don't know if that's something that user's would want to do but it feels something that we could add on further PRs... what do you think, @yanokwa @ln?

Reply:
I agree that it should be possible to only override the date or the PEM file. Will file as a separate issue.